boa/cli/mambabuild.py

```diff
@@ -27,45 +27,60 @@
 
 solver_map = {}
 
+# e.g. package-1.2.3-h5487548_0
+dashed_spec_pattern = r"([^ ]+)-([^- ]+)-([^- ]+)"
+# e.g. package 1.2.8.*
+conda_build_spec_pattern = r"([^ ]+)(?:\ ([^ ]+))?(?:\ ([^ ]+))?"
+
+problem_re = re.compile(
+    rf"""
+        ^(?:\ *-\ +)+
+        (?:
+            (?:
+                package
+                \ {dashed_spec_pattern}
+                \ requires
+                \ {conda_build_spec_pattern}
+                ,\ but\ none\ of\ the\ providers\ can\ be\ installed
+            ) | (?:
+                package
+                \ {dashed_spec_pattern}
+                \ has\ constraint
+                \ .*
```

Comment:
N.B.: This is `.*` rather than `{conda_build_spec_pattern}` since `parse_problems` (before and after this PR) only outputs one value per `name`.
This is to not overshadow, e.g.,
`MatchSpec("pkg==123=abc")` by `MatchSpec("pkg=456")` for some
`"other-pkg has constraint pkg 456.* conflicting with pkg-123-abc"` string.